src/services/scriptingservice.h

```diff
@@ -47,7 +49,10 @@ class ScriptingService : public QObject {
     bool methodExists(const QString &methodName) const;
     static bool validateScript(const Script &script, QString &errorMessage);
     Q_INVOKABLE bool startDetachedProcess(const QString &executablePath,
-                                          const QStringList &parameters);
+                                          const QStringList &parameters,
+                                          const QString &identifier = QString(),
+                                          const int index = 0,
```

Comment:
Wouldn't it be better to use a `QVariant callbackParameter`, then a script could pass everything

Reply:
Could you please elaborate?

Reply:
What are you using the index for? Is that something that is passed by the script?

Reply:
Yes, e. g. the index of a for-loop when starting a bunch of commands associated with one callbackIdentifier.

Reply:
if you use `QVariant` the script could also pass a string, float, bool or a stringlist... the variable isn't interpreted or modified in C++, right? just passed along...

Reply:
Ah you are talking about generalizing the index variable, good point!

Reply:
exactly